main.c

```diff
@@ -1,23 +1,32 @@
 #include <stdio.h>
+#include <stdlib.h>
 #include <stdbool.h>
 #include "src/lexer.h"
 #include "src/parser.h"
+#include "src/utils.h"
 
 int main(int argc, char *argv[])
 {
     char * file_name;
-    char output_file_name[] = "some_file.md";
+    char * output_file;
     if(!(file_name = argv[1])) {
-        return 0;
+        return EXIT_FAILURE;
     }
 
-    int i;
-    Lexer *lexer = lex_file(file_name);
-    parse(lexer, output_file_name);
+    // Is the path directory?
+    if (is_directory(file_name)) {
+        printf("This tool can only be used in files!\n");
+        return EXIT_FAILURE;
+    }
 
+    // Ajust the extension of the file
+    output_file = change_extension(file_name, ".html");
 
+    Lexer *lexer = lex_file(file_name);
+    parse(lexer, output_file);
+    free(output_file);
```

Comment:
you're doing the kernel's job, when the program exits kernel already free whole memory space, you don't need to free things right before exit

Reply:
I wasnt sure if this was the case so i left it in. i will remove it.